homeassistant/components/bayesian/binary_sensor.py

```diff
@@ -424,25 +427,32 @@ def _build_observations_by_template(self) -> dict[Template, list[Observation]]:
 
         return observations_by_template
 
-    def _process_numeric_state(self, entity_observation: Observation) -> bool | None:
+    def _process_numeric_state(
+        self, entity_observation: Observation, multi: bool = False
+    ) -> bool | None:
         """Return True if numeric condition is met, return False if not, return None otherwise."""
         entity = entity_observation.entity_id
 
         try:
             if condition.state(self.hass, entity, [STATE_UNKNOWN, STATE_UNAVAILABLE]):
                 return None
-            return condition.async_numeric_state(
+            result = condition.async_numeric_state(
```

Comment:
Hitting a problem where numeric state with above and below always looks for `>=` or `<=` so when the sensor state is on the threshold it always evaluates to `False`
```yaml
binary_sensor:
  name: "Heat On"
  platform: "bayesian"
  prior: 0.2
  probability_threshold: 0.9
  observations:
    - platform: "numeric_state"
      entity_id: "sensor.outside_air_temperature_fahrenheit"
      prob_given_true: 0.95
      prob_given_false: 0.05
      below: 50
    - platform: "numeric_state"
      entity_id: "sensor.outside_air_temperature_fahrenheit"
      prob_given_true: 0.01
      prob_given_false: 0.01
      above: 50
      below: 100
    - platform: "numeric_state"
      entity_id: "sensor.outside_air_temperature_fahrenheit"
      prob_given_true: 0.01
      prob_given_false: 0.02
      above: 100
```
Setting state to 50 will cause none of them to be seen